analyzers/src/SonarAnalyzer.SourceGenerator/RuleCatalogGenerator.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Newtonsoft.Json.Linq;
+
+namespace SonarAnalyzer.SourceGenerator
+{
+    [Generator]
+    [ExcludeFromCodeCoverage]
```

Comment:
It's not worth the effort. It would complicate the code without bringing additional value. It is tested by being able to build the solution and the product is also tested by a bunch of specific UTs in `RuleCatalogTest`, `DiagnosticDescriptorBuilderTest`, `RuleTest`